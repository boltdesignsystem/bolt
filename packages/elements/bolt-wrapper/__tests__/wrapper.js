// Refer to https://github.com/boltdesignsystem/bolt/wiki/Jest-Test:-Example-Jest-Test for more testing examples
import { render, stopServer } from '../../../testing/testing-helpers';
// import schema from '../wrapper.schema';
// const { [PROP KEY], [PROP KEY]... } = schema.properties;
let page, fixtures;

afterAll(async () => {
  await stopServer();
  await page.close();
});

beforeEach(async () => {
  await page.evaluate(() => {
    document.body.innerHTML = '';
  });
});

beforeAll(async () => {
  page = await global.__BROWSER__.newPage();
  await page.goto('http://127.0.0.1:4444/', {
    timeout: 0,
  });

  const defaultData = {
    content: 'This is a wrapper',
  };

  fixtures = {
    defaultData,
  };
});

describe('Bolt Wrapper element', () => {
  test('default', async () => {
    const results = await render(
      '@bolt-elements-wrapper/wrapper.twig',
      {
        ...fixtures.defaultData,
      },
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});

// describe('Bolt Wrapper prop -', () => {
//   // Target each of the schema keys with the following pattern
//   [PROP KEY].enum.forEach(async (option) => {
//     test(`[PROP KEY] items: ${option}`, async () => {
//       const results = await render(
//         '@bolt-components-wrapper/wrapper.twig',
//         {
//           ...fixtures.defaultData,
//           [PROP KEY]: option,
//         }
//       );

//       await expect(results.ok).toBe(true);
//       await expect(results.html).toMatchSnapshot();
//     });
//   });
// });
