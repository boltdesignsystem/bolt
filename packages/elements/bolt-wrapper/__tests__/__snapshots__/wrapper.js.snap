// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt Wrapper element default 1`] = `
<div class="e-bolt-wrapper">
  This is a wrapper
</div>
`;
